composition-js/src/merging/merge.ts

```diff
@@ -2058,6 +2072,69 @@ class Merger {
       });
     }
   }
+  
+  private removeOverriddenFieldFromFieldSet <T extends FieldDefinition<ObjectType | InterfaceType> | InputFieldDefinition>(
```

Comment:
nit: I think the naming is somewhat confusing as this is specifically for `@provides` but function name doesnt mention it at all